### PR TITLE
feat: add ability to switch app mode

### PR DESCRIPTION
**Pull request by kyranjamie**
This PR scaffolds some of the new routes we'll need for the main app. It also introduces the need for a `.env` file to toggle between the two _App start modes_, either `live` or `prelaunch`.

Expo Router is kind of confusing, and was stumped by this issue https://github.com/expo/router/issues/723, so setting the initial route involves first going to one, then navigating. Not ideal but none of the `initialRouteName` settings work.

You can long press the Bitcoin icon on the welcome screen to get to the main app page.

https://github.com/leather-wallet/mono/assets/1618764/ad3e08a2-ed1e-489c-8c9a-ca531153f575

## Summary by CodeRabbit

- **New Features**
  - Introduced a React component for the home screen in the mobile app.
  - Added new components: `SplashScreenGuard`, `WaitingListPage`, `LeatherSplash`, and `Spinner`.
  - Added utility function `whenAppStartMode` for enforcing app start modes.

- **Improvements**
  - Updated mobile app `README.md` to use `pnpm` instead of `npm`.
  - Simplified exports configuration in the UI package `package.json`.
  - Enhanced localization files with updated message locations and translations.
  - Updated import paths and refactored multiple components for better organization.

- **Bug Fixes**
  - Corrected error messages in the email submission functionality.
  - Fixed import path issues for `Spinner` component.

- **Documentation**
  - Added instructions for setting app mode using an `.env` file in the mobile app `README.md`.

- **Refactor**
  - Replaced `Stack` with `Slot` in the app's layout.
  - Renamed and refactored multiple components and hooks for improved clarity and functionality.

- **Chores**
  - Updated `.gitignore` and `.ls-lint.yml` configurations.

### CODE REVIEW

File: apps/mobile/src/app/waiting-list/waiting-list.tsx [hunk 1]
```diff
@@ -346,7 +349,12 @@ export function WelcomeScreen() {
             <Animated.View
               style={[{ position: 'absolute', transform: [{ rotateZ: '100deg' }] }, btcStyle]}
             >
-              <Image style={{ width: 73, height: 73 }} source={require('@/assets/Pixel BTC.png')} />
+              <Pressable onLongPress={() => onHiddenPressAction()}>
+                <Image
+                  style={{ width: 73, height: 73 }}
+                  source={require('@/assets/Pixel BTC.png')}
```
Comment (pete-watters) [inline apps/mobile/src/app/waiting-list/waiting-list.tsx:355]: Does this work with a space in the filename? `assets/Pixel BTC.png')`

File: packages/ui/package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
       "import": "./dist-all/web.js",
       "require": "./dist-all/web.js"
     },
-    "./native": "./dist-all/native.js"
+    "./native": "./dist-native/native.js"
```
Comment (pete-watters) [inline packages/ui/package.json:33]: 👍 apologies if my change caused you an issue.

**Review by pete-watters — APPROVED** [review]
👍 nice!

**Review by fbwoolf — APPROVED** [review]

File: .ls-lint.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ ls:
   .ts: regex:[a-z0-9\-\._]+
   .tsx: regex:[a-z0-9\-\._]+
   .js: regex:[a-z0-9\-\._]+
-  .dir: regex:[a-z0-9\-\._]+
+  .dir: regex:[a-z0-9\(\)\-\._]+
```
Comment (kyranjamie) [inline .ls-lint.yml:5]: This is needed for Expo Router's file based routing conventions

File: apps/mobile/src/utils/when-app-start-mode.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+export type AppStartMode = (typeof appStartModes)[number];
+
+type AppStartModeMap<T = any> = Record<AppStartMode, T>;
+
+export function whenAppStartMode<K extends AppStartMode>(mode: K) {
+  if (!appStartModes.includes(mode))
+    throw new Error('App start mode must be one of: ' + appStartModes.join(', '));
+
+  return <M extends AppStartModeMap>(appStartModeMap: M): M[K] => appStartModeMap[mode];
+}
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/utils/when-app-start-mode.ts:12]: I disagree here, typescript string literals offer all the same type safety of enums, and all the developer-friendly features (autocomplete), without the need for an additional import of the enum.

Agree with this analysis https://nikolasbarwicki.com/articles/enums-vs-string-literal-types/

File: apps/mobile/src/app/(home)/use-force-initial-app-start-route.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { router, useFocusEffect } from 'expo-router';
+
+let hasRun = false;
+
+const startRoute = whenAppStartMode(process.env.EXPO_PUBLIC_APP_START_MODE as AppStartMode)({
+  live: '/',
+  prelaunch: '/waiting-list',
+});
+
+export function useForceInitialAppStartRoute() {
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/app/(home)/use-force-initial-app-start-route.ts:13]: Will try with the redirect component

**Comment by kyranjamie** [comment]
Thanks for review @edgarkhanzadian. Not sure why you get the flash of the other view, and I didn't in my demo

File: apps/mobile/src/app/(home)/use-force-initial-app-start-route.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/app/(home)/use-force-initial-app-start-route.ts]: Yep, worked, better solution.

File: apps/mobile/src/utils/when-app-start-mode.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/utils/when-app-start-mode.ts:12]: I'm not so opinionated on these two approaches. The former is cleaner/less repetitive imo. No material difference using `.includes()` vs `appStartModes[mode]`.

**Pull request merged**